base_report_docx/report/report_docx.py

```diff
@@ -226,9 +235,19 @@ def _convert_docx_from_template(
             template_path, base64.b64decode(action.template_file.datas))
 
         doc = DocxTemplate(template_path)
+        # this is a hook to perform additional operations on the DOC file
+        # before it's rendered
+        self.on_doc_render(doc, data, context)
         doc.render(data)
         doc.save(convert_path)
 
+    def on_doc_render(doc, data, context):
+        """
+            Override this function to do additions thing on the Docx file that
```

Comment:
Replace with the same comment as the one above:
"Override this function to perform additional operations on the DOC file before it's rendered."